fastscanner/adapters/rest/scanner.py

```diff
@@ -0,0 +1,136 @@
+
+    async def handle(self, symbol: str, new_row: pd.Series, passed: bool) -> pd.Series:
+        if not passed:
+            return new_row
+
+        scan_time = datetime.now().strftime("%H:%M")
+        candle = new_row.to_dict()
+
+        scan_time = new_row.name.strftime("%H:%M")  # type: ignore
+        message = ScannerMessage(
+            symbol=symbol,
+            scan_time=scan_time,
+            scanner_id=self._scanner_id,
+            candle=candle,
+        )
+
+        await self._send_message(message)
+
+        return new_row
+
+    def set_scanner_id(self, scanner_id: str):
+        self._scanner_id = scanner_id
+
+    async def _send_message(self, message: ScannerMessage):
+        try:
+            message_json = message.model_dump_json()
+            logger.info(f"Sending scanner result: {message_json}")
+            await self._websocket.send_text(message_json)
+        except Exception as e:
+            logger.error(f"Failed to send websocket message: {e}")
+
+
+def _parse_known_parameters(params: Dict[str, Any]) -> Dict[str, Any]:
+    """Parse known parameters and convert them to appropriate types."""
+    processed_params = params.copy()
+
+    if "start_time" in processed_params:
+        processed_params["start_time"] = time.fromisoformat(
+            processed_params["start_time"]
+        )
+
+    if "end_time" in processed_params:
+        processed_params["end_time"] = time.fromisoformat(processed_params["end_time"])
+
+    return processed_params
+
+
+@router.websocket("")
+async def websocket_realtime_scanner(websocket: WebSocket):
+    await websocket.accept()
+    scanner_id = None
+    service = None
+
+    polygon = PolygonCandlesProvider(config.POLYGON_BASE_URL, config.POLYGON_API_KEY)
+    candles = PartitionedCSVCandlesProvider(polygon)
+    channel = RedisChannel(
+        unix_socket_path=config.UNIX_SOCKET_PATH,
+        host=config.REDIS_DB_HOST,
+        port=config.REDIS_DB_PORT,
+        password=None,
+        db=0,
+    )
+    service = ScannerService(candles=candles, channel=channel, symbols_provider=polygon)
+
+    data = await websocket.receive_text()
+    request_data = json.loads(data)
+
+    scanner_request = ScannerRequest(**request_data)
+    processed_params = _parse_known_parameters(scanner_request.params)
+
+    scanner_params = ScannerParams(type_=scanner_request.type, params=processed_params)
+
+    handler = WebSocketScannerHandler(websocket, "")
+
+    freq = processed_params["freq"]
+
+    scanner_id = await service.subscribe_realtime(
+        params=scanner_params, handler=handler, freq=freq
+    )
+
+    handler.set_scanner_id(scanner_id)
+
+    response = ScannerResponse(scanner_id=scanner_id)
+    logger.info(f"response is as follows: {response}")
+    await websocket.send_text(response.model_dump_json())
+
+    logger.info(f"Started scanner with ID: {scanner_id}")
+
+    while True:
+        await websocket.receive_text()
+
+    if scanner_id and service:
```

Comment:
I still wonder why you add these random if conditionals in the code. When is scanner_id or service gonna be None? We need to start getting rid of this habit.